crates/dwallet-mpc-centralized-party/src/lib.rs

```diff
@@ -558,6 +575,76 @@ fn protocol_public_parameters(
+    let reconfiguration_dkg_public_output: VersionedDecryptionKeyReconfigurationOutput =
+        bcs::from_bytes(&reconfiguration_dkg_public_output)?;
+
+    // Every member has a voting power of 1 in the current Move code.
+    let access_structure = WeightedThresholdAccessStructure::new(
+        quorum_threshold as Weight,
+        (1..=committee_size).map(|i| (i as PartyID, 1)).collect(),
+    )?;
+
+    match &reconfiguration_dkg_public_output {
+        VersionedDecryptionKeyReconfigurationOutput::V1(public_output_bytes) => {
+            let public_output: <class_groups::reconfiguration::Secp256k1Party as mpc::Party>::PublicOutput =
+                bcs::from_bytes(public_output_bytes)?;
+
+            let decryption_key_share_public_parameters = public_output
+                .default_decryption_key_share_public_parameters::<secp256k1::GroupElement>(
+                    &access_structure,
+                )?;
+            let encryption_scheme_public_parameters = decryption_key_share_public_parameters
+                .encryption_scheme_public_parameters
+                .clone();
+
+            let neutral_group_value =
+                group::secp256k1::GroupElement::neutral_from_public_parameters(
+                    &group::secp256k1::group_element::PublicParameters::default(),
+                )
+                .map_err(twopc_mpc::Error::from)?
+                .value();
+            let neutral_ciphertext_value =
+                ::class_groups::CiphertextSpaceGroupElement::neutral_from_public_parameters(
+                    encryption_scheme_public_parameters.ciphertext_space_public_parameters(),
+                )
+                .map_err(twopc_mpc::Error::from)?
+                .value();
+
+            let protocol_public_parameters = twopc_mpc::ProtocolPublicParameters::new::<
+                { secp256k1::SCALAR_LIMBS },
+                { FUNDAMENTAL_DISCRIMINANT_LIMBS },
+                { NON_FUNDAMENTAL_DISCRIMINANT_LIMBS },
+                secp256k1::GroupElement,
+            >(
+                neutral_group_value,
+                neutral_group_value,
+                neutral_ciphertext_value,
+                neutral_ciphertext_value,
+                decryption_key_share_public_parameters
+                    .encryption_scheme_public_parameters
+                    .clone(),
+            );
+
+            Ok(protocol_public_parameters)
+        }
+        VersionedDecryptionKeyReconfigurationOutput::V2(public_output_bytes) => {
+            let public_output: <twopc_mpc::decentralized_party::reconfiguration::Party as mpc::Party>::PublicOutput =
+                bcs::from_bytes(public_output_bytes)?;
+            let secp256k1_protocol_public_parameters =
+                twopc_mpc::decentralized_party::reconfiguration::PublicOutput::secp256k1_protocol_public_parameters(
+                    &public_output,
+                )?;
+
+            Ok(secp256k1_protocol_public_parameters)
```

Comment:
We have functions that are in Yael's latest PRs for the centralized party that creates the public input for protocols generically.

I know we want to cache things in the user side, we still should, but maybe we should do that generically too - or just cache for all the protocols.